### ISSUE TITLE
Create messages to prevent self-harm?

### ISSUE THREAD

**Issue by p0bs** [tidymodels/recipes#414]
Having chatted briefly about this at the Applied Machine Learning course in London, you thought it made sense for me to raise an issue on this point.

Specifically, given the opinionated nature of `recipes`, I wondered if you could create a warning or message in `tidymodels` (and, in particular, `recipes`) if a user seems to be on the verge of self-harm. For example, if they were about to use the test set when they should be using the training set. 

Whilst there might be more elegant ways, it could be done by scraping the string of the object name to see if it included (a variant of) the word 'test' in some way. 

I know that I'm paranoid about doing something like this by accident and I sense that I'm not alone on this.

Just thought it worth suggesting it as a possible idea.

**Comment by topepo** [comment]
> I know that I'm paranoid about doing something like this by accident and I sense that I'm not alone on this.

You're not. 

I've played around with this and don't have a way that wouldn't lead to a lot of false positives (at least, based off of the name of the object).

I'll close this but please comment if anyone can think of a better method for catching these issues.
